segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -308,10 +308,15 @@ public static ChunkedSegmentStorage getReference(Storage storage) {
      *                        throws StorageMetadataException In case of any chunk metadata store related errors.
      */
     private CompletableFuture<Void> claimOwnership(MetadataTransaction txn, SegmentMetadata segmentMetadata) {
+        Preconditions.checkState(!segmentMetadata.isStorageSystemSegment(), "claimOwnership called on system segment %s", segmentMetadata);
```

Comment:
Question: the reason why we still keep all the reconciliation stuff here is to make this change backwards compatible, right? Because we may be upgrading from a Pravega cluster with `lazyCommit=true` to a a cluster with this change, in which case we need to also reconcile as the data/metadata left in the system may have not been atomically updated, right? If so, why system segments are not reconciled?

Reply:
1. Yes , we have kept the reconciliation logic to handle upgrade.
2. System segments are always recovered as first step during SLTS bootstrap via replaying of journal entries. Because they are reconciled every time container restarts , `claimOwnership` is never actually called on them (the metadata is already updated during bootstrap.)
3. SystemJournal logic takes care of migration scenarios and unit tests cover these scenarios separately